File: backend/app/core/currency.py
"""Rwanda Franc — platform-wide currency for budgets and training costs."""

from __future__ import annotations

CURRENCY_CODE = "FRW"


def format_frw(amount: int | float | None) -> str:
    if amount is None:
        return "—"
    try:
        n = int(round(float(amount)))
    except (TypeError, ValueError):
        return "—"
    return f"{CURRENCY_CODE} {n:,}"

File: frontend/src/utils/currency.js
/** Rwanda Franc — platform-wide currency display. */
export const CURRENCY_CODE = "FRW";

export function formatFrw(amount, options = {}) {
  const { fallback = "—" } = options;
  if (amount == null || amount === "") return fallback;
  const n = Number(amount);
  if (!Number.isFinite(n)) return fallback;
  const formatted = n.toLocaleString("en-RW", { maximumFractionDigits: 0 });
  return `${CURRENCY_CODE} ${formatted}`;
}
